internal/fs/local_file_test.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// A collection of tests for a file system where we do not attempt to write to
+// the file system at all. Rather we set up contents in a GCS bucket out of
+// band, wait for them to be available, and then read them via the file system.
+
+package fs_test
+
+import (
+	"errors"
+	"os"
+	"path"
+
+	"github.com/googlecloudplatform/gcsfuse/internal/config"
+	"github.com/jacobsa/gcloud/gcs"
+	"github.com/jacobsa/gcloud/gcs/gcsutil"
+	. "github.com/jacobsa/ogletest"
+)
+
+////////////////////////////////////////////////////////////////////////
+// Boilerplate
+////////////////////////////////////////////////////////////////////////
+
+type LocalFileTest struct {
+	fsTest
+}
+
+func init() {
+	RegisterTestSuite(&LocalFileTest{})
+}
+
+func (t *LocalFileTest) SetUpTestSuite() {
+	t.serverCfg.ImplicitDirectories = true
+	t.serverCfg.MountConfig = &config.MountConfig{
+		WriteConfig: config.WriteConfig{CreateEmptyFile: false},
+	}
+	t.fsTest.SetUpTestSuite()
+}
+
+////////////////////////////////////////////////////////////////////////
+// Tests
+////////////////////////////////////////////////////////////////////////
+
+func (t *LocalFileTest) TestReadDirWithEmptyLocalFile() {
```

Comment:
can you add test case for listing non-empty files and checkinng if the size is returned correctly.
This is our main usecase. The purpose of supporting ls is to show the size of the file.

Reply:
This test case requires lookUpInode to be supported for local files. I'll add this test in a separate PR.

Reply:
added a TODO for the same. [link](https://github.com/GoogleCloudPlatform/gcsfuse/pull/1287/files#diff-7800585ad28fc6c239f5a8d71b053eab502f7f40d5c0c7d6b794d4a4c3386afcR211)

Reply:
Added the test